nil pointer dereference during compaction

Repro is with Zed commit f5350b0.

I've only been able to reproduce this issue by having some records that lack the pool key mixed in with the ones that do. Steps:

1. Start a `zed serve`

```
$ zed -version
Version: v1.2.0-102-gf5350b0c

$ rm -rf lake && zed serve -lake lake -log.level debug
{"level":"info","ts":1667508957.307296,"msg":"Open files limit raised","limit":61440}
...
```

2. Create a pool with small object size.

```
$ zed create -S 10KB zeek
pool created: zeek 2H3KUgq5lEO5udUEeoqeieLVnc3
```

3. Start a `zed monitor manage` with intent to compact data as it accumulates. I notice that HTTP 500 error at startup, but that symptom looks similar to existing issue #3029 and I'm guessing is unrelated.

```
$ cat manage.yaml 
coldthresh: 0s

$ zed manage monitor -config manage.yaml
{"level":"error","ts":1667508346.559808,"logger":"pool","msg":"scan error","name":"zeek","id":"2H3KUgq5lEO5udUEeoqeieLVnc3","branch":"main","error":"status code 500: unknown pool metadata type: \"objects\"","stacktrace":"github.com/brimdata/zed/cmd/zed/manage/lakemanage.(*poolMonitor).run.func1\n\t/Users/phil/work/zed/cmd/zed/manage/lakemanage/monitor.go:169"}
...
```

4. Unpack the attached [data.tgz](https://github.com/brimdata/zed/files/9932874/data.tgz) and load using the following loop. As you can see, there's one `ts`-free record loaded after each 100 of the regular `conn` records.

```
$ tar xvf data.tgz
$ NUM="1" ; while [ "$NUM" -le 100 ]; do zed load $NUM.zson; NUM=$(expr $NUM + 1); done; echo '{"_path":"loaded_scripts","_write_ts":"1970-01-01T00:00:00.000000Z","name":"/usr/local/zeek-5.0.2/share/zeek/base/init-bare.zeek"}' | zed load -
```

It seems that if I repeat that looped load three times in a row, consistently on the third time through, the `zed manage monitor` dies with the following stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x165aa83]

goroutine 3382 [running]:
github.com/brimdata/zed/runtime/expr/extent.(*Generic).Overlaps(0x0, 0xc000193620?, 0xc000324ba0?)
	/Users/phil/work/zed/runtime/expr/extent/span.go:90 +0x23
github.com/brimdata/zed/cmd/zed/manage/lakemanage.(*Run).SizeUncoveredBy(0xc0000fecb8?, {0x1d824e0, 0x0})
	/Users/phil/work/zed/cmd/zed/manage/lakemanage/run.go:41 +0x7b
github.com/brimdata/zed/cmd/zed/manage/lakemanage.Scan.func1({{0x1d824e0, 0xc000388d20}, 0xc00007bfe0, {0xc0002807a0, 0x2, 0x2}}, {0x1d824e0?, 0x0?})
	/Users/phil/work/zed/cmd/zed/manage/lakemanage/scan.go:27 +0xa7
github.com/brimdata/zed/cmd/zed/manage/lakemanage.Scan({0x1d7d1f0?, 0xc0000f4600?}, {0x1d716c0, 0xc0000b8510}, 0xc0000d41e0, 0x0, 0xc00015e5b0?)
	/Users/phil/work/zed/cmd/zed/manage/lakemanage/scan.go:68 +0x453
github.com/brimdata/zed/cmd/zed/manage/lakemanage.(*poolMonitor).scan.func1()
	/Users/phil/work/zed/cmd/zed/manage/lakemanage/monitor.go:193 +0x57
created by github.com/brimdata/zed/cmd/zed/manage/lakemanage.(*poolMonitor).scan
	/Users/phil/work/zed/cmd/zed/manage/lakemanage/monitor.go:192 +0x212
```

--- a/lake/data/object.go
+++ b/lake/data/object.go
@@ -121,10 +121,6 @@ func NewObject() Object {
 }
 
 func (o Object) Span(order order.Which) *extent.Generic {
-	if o.First.Bytes == nil || o.Last.Bytes == nil {
-		//XXX
-		return nil
-	}
 	return extent.NewGenericFromOrder(o.First, o.Last, order)
 }
 

